go.mod

```diff
@@ -421,7 +421,7 @@ require (
 	github.com/go-openapi/validate v0.23.0 // indirect
 	github.com/go-redis/redis/v8 v8.11.5 // indirect
 	github.com/go-resty/resty/v2 v2.13.1 // indirect
-	github.com/go-viper/mapstructure/v2 v2.0.0-alpha.1 // indirect
+	github.com/go-viper/mapstructure/v2 v2.0.0 // indirect
```

Comment:
The error is from this line, how the mapstructure reports errors has changed. You need to update the verbiage in the corrupt_config.diags to match the new error it reports.

Reply:
Nice find, thanks!